src/node_dotenv.cc

```diff
@@ -13,29 +13,22 @@ using v8::String;
 
 std::vector<std::string> Dotenv::GetPathFromArgs(
     const std::vector<std::string>& args) {
-  const auto find_match = [](const std::string& arg) {
-    const std::string_view flag = "--env-file";
-    return strncmp(arg.c_str(), flag.data(), flag.size()) == 0;
```

Comment:
We don't need this whole change. Changing only the current highlighted line with this, and later check if `find_match == "--"` and return early, would fix the bug.

```
return arg == "--" || arg.starts_with("--env-file");
```

Reply:
I made those changes, tests are now failing :-/